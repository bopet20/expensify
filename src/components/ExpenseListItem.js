import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <p className="list-item__title">{description}</p>
      <span className="list-item__subtitle">{moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <p className="list-item__data">{numeral(amount/100).format('$0,0.00')}</p>
  </Link>
)

export default ExpenseListItem