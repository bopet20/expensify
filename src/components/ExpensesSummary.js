import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const formattedAmount = numeral(expensesTotal/100).format('$0,0.00')

  return (
    <div>
      <h2>Viewing {expenseCount} expense{expenseCount === 1 ? '' : 's'} totalling {formattedAmount}</h2>
    </div>
  )
}

const mapStateToProps = ({ expenses, filters }) => {
  const selectedExpenses = selectExpenses(expenses, filters)

  return {
    expenseCount: selectedExpenses.length,
    expensesTotal: selectExpensesTotal(selectedExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
