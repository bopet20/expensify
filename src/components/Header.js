import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <nav>
      <NavLink exact to="/" activeClassName="is-active">Home</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create</NavLink>
    </nav>
  </header>
)

export default Header