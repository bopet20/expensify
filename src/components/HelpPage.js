import React from 'react'

const HelpPage = () => (
  <div>
    help page component
  </div>
)

export default HelpPage