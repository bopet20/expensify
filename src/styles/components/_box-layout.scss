.box-layout {
  align-items: center;
  background-color: #ccdfce;
  background-image: $bg-image;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;

  &__box {
    background: fade-out(white, .2);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}
