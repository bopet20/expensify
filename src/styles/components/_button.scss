.button {
  cursor: pointer;
  background-color: $wintergreen;
  padding: $s-size;
  border: none;
  color: white;
  border-radius: 3px;
  font-size: $font-size-large;
  font-weight: 300;
  display:inline-block;
  text-decoration: none;
  line-height: 1;

  &--google {
    background-color: $google-red;
  }

  &--link {
    background: none;

    &--secondary {
      color: #888888;
      font-size: $font-size-small;
      padding: 0 0 0 6px;
    }
  }

  &--secondary {
    background: #888888;
  }
}
