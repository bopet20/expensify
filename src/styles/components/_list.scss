.list-header {
  background: $wintergreen;
  border: 1px solid $dark-wintergreen;
  color: white;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-body {
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}

.list-item {
  border: 1px solid $dark-wintergreen;
  border-top: none;
  color: $dark-green;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background .3s ease;

  &:hover {
    background: $secondary-green;
  }

  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }

  &__title {
    font-size: $font-size-large;
    font-weight: bold;
    margin: 0;
    word-break: break-all;
  }

  &__subtitle {
    color: $dark-wintergreen;
    font-size: $font-size-small;
  }

  &__data {
    font-size: $font-size-large;
    font-weight: bold;
    margin: $s-size 0 0 0;

    @media (min-width: $desktop-breakpoint) {
      margin: 0;
      padding-left: $s-size;
    }
  }

  &--message {
    align-items: center;
    color: $dark-wintergreen;
    justify-content: center;
    padding: $m-size;

    &:hover {
      background: none;
    }
  }
}
