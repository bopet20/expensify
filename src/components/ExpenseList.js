import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Expenses</div>
      <div className="show-for-desktop">Expense</div>
      <div className="show-for-desktop">Amount</div>
    </div>
    <div className="list-body">
      {
        props.expenses.length === 0 ? (
          <span className="list-item list-item--message">No expenses</span>
        ) : (
          props.expenses.map((expense) => (
            <ExpenseListItem {...expense} key={expense.id} />
          ))
        )
      }
    </div>
  </div>
)

const mapStatetoProps = ({ expenses, filters }) => {
  return {
    expenses: selectExpenses(expenses, filters)
  }
}

export default connect(mapStatetoProps)(ExpenseList)
