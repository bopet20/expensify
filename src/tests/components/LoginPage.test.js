import { LoginPage } from '../../components/LoginPage'

describe('<LoginPage />', () => {
  let startLogin, wrapper
  beforeEach(() => {
    startLogin = jest.fn()
    wrapper = shallow(<LoginPage startLogin={startLogin}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('handles login with onClick', () => {
    wrapper.find('button').simulate('click')
    expect(startLogin).toHaveBeenCalled()
  })
})