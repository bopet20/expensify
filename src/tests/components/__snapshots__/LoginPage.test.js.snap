// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> renders correctly 1`] = `
<Fragment>
  <button
    onClick={[MockFunction]}
    type="submit"
  >
    Login
  </button>
</Fragment>
`;
